python/lsst/ts/scheduler/too_client.py

```diff
@@ -0,0 +1,242 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["TooClient"]
+
+import dataclasses
+import logging
+
+import healpy as hp
+import lsst_efd_client
+import numpy as np
+from astropy import units
+from astropy.time import Time, TimeDelta
+from lsst.ts.utils import index_generator
+from numpy.typing import NDArray
+
+
+@dataclasses.dataclass
+class TooAlert:
+    """Data structure with the Target of opportunity alert payload."""
+
+    source: str
+    """A unique identifier for this event"""
+
+    tooid: int
+    """Unique identifier of the ToO."""
+
+    instrument: list[str]
+    """A list of names of instruments responsible for the
+    observations which lead to issuing this alert"""
+
+    alert_type: str
+    """The categorization of this alert,
+    defined by which filter criteria it passed"""
+
+    event_trigger_timestamp: str
+    """The UTC time of the event described by the alert
+    in ISO-8601 format"""
+
+    reward_map: NDArray[np.bool]
+    """The pixels of a HEALPix map, in the nested ordering,
+    with binary values indicating whether they should be
+    targeted for observation"""
+
+    reward_map_nside: int
+    """The n_side parameter describing the resolution of the
+    map data in reward_map"""
+
+    is_test: bool
+    """A flag indicating whether the event is a test or
+    simulated alert"""
+
+    is_update: bool
+    """A flag indicating that this is an update to a previous
+    version of the same event"""
+
+
+class TooClient:
+    """Handle Target of Oportunity alerts.
+
+    Parameters
+    ----------
+    topic_name : `str`
+        The name of the topic with the ToO data in the EFD.
+    delta_time : `float`
+        How long in the past to look for ToO alerts?
+        Older alerts will be ignored.
+    efd_name : `str`
+        Name of the EFD instance alerts should be queried from.
+    db_name : `str`, optional
+        The database name where the topic is written (Default, efd).
+    log : `logging.Logger`, optional
+        Logger class.
+    """
+
+    def __init__(
+        self,
+        topic_name: str,
+        delta_time: float,
+        efd_name: str,
+        db_name: str = "efd",
+        log: logging.Logger | None = None,
+    ) -> None:
+
+        self.log = (
+            logging.getLogger(type(self).__name__)
+            if log is None
+            else log.getChild(type(self).__name__)
+        )
+        self.topic_name = topic_name
+        self.delta_time = delta_time
+        self.efd_client = lsst_efd_client.EfdClient(
+            efd_name,
+            db_name=db_name,
+        )
+
+        self.too_alerts: dict[str, TooAlert] = dict()
+
+        self._index_generator = index_generator()
+
+    def get_initial_query_parameters(self) -> list[str]:
```

Comment:
Do you really need a method for this? This could be a constant, right? Or do you plan to mock this in unit tests?